plugins-experiments/src/main/kotlin/org/gradle/kotlin/dsl/experiments/plugins/GradleKotlinDslKtlintConventionPlugin.kt

```diff
@@ -0,0 +1,136 @@
+package org.gradle.kotlin.dsl.experiments.plugins
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.execution.TaskExecutionListener
+import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.SourceSet
+import org.gradle.api.tasks.TaskState
+
+import org.gradle.cache.internal.GeneratedGradleJarCache
+import org.gradle.internal.hash.Hashing
+import org.gradle.internal.logging.ConsoleRenderer
+
+import org.jlleitschuh.gradle.ktlint.KtlintExtension
+import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
+
+import org.gradle.kotlin.dsl.*
+import org.gradle.kotlin.dsl.support.serviceOf
+
+
+private
+const val rulesetJarName = "gradle-kotlin-dsl-ruleset.jar"
+
+
+open class GradleKotlinDslKtlintConventionPlugin : Plugin<Project> {
+
+    override fun apply(project: Project): Unit = project.run {
+
+        plugins.apply("org.jlleitschuh.gradle.ktlint")
+
+        configure<KtlintExtension> {
+            version = DefaultVersions.ktlint
+            reporters = arrayOf(ReporterType.PLAIN)
+        }
+
+        val ktlint by configurations.creating {
+            exclude(module = "ktlint-ruleset-standard")
+        }
+
+        dependencies {
+            ktlint(files(gradleKotlinDslKtlintRulesetJar()))
+        }
+
+        afterEvaluate {
+            fixKtlintTasks()
+        }
+    }
+
+
+    private
+    fun Project.gradleKotlinDslKtlintRulesetJar() = provider {
+        val ruleset = this@GradleKotlinDslKtlintConventionPlugin.javaClass
+            .getResourceAsStream(rulesetJarName)!!.use { it.readBytes() }
+        val cacheKey = Hashing.md5().hashBytes(ruleset).toString()
```

Comment:
:x: Wouldn't the kotlin-dsl version or plugin version work as a cheaper cache key here considering that `GeneratedGradleJarCache` is already Gradle version specific?

Reply:
:x: Alternatively, the cache key could be another resource.